graphprot/Metrics.py

```diff
@@ -8,68 +8,107 @@
 def get_boolean(vals, threshold, target):
         
         if target == 'fnat' or target == 'bin':
-                vals_bool = [1 if x >= threshold else 0 for x in vals]
+                vals_bool = [1 if x > threshold else 0 for x in vals]
         else: 
                 vals_bool = [1 if x < threshold else 0 for x in vals]
         
         return vals_bool
 
-def get_comparison(prediction, ground_truth):
+
+def get_comparison(prediction, ground_truth, binary=True, classes=[0,1]):
         
-    CM = confusion_matrix(ground_truth, prediction)
+    CM = confusion_matrix(ground_truth, prediction, labels = classes)
     
     FP = CM.sum(axis=0) - np.diag(CM)
     FN = CM.sum(axis=1) - np.diag(CM)
     TP = np.diag(CM)
     TN = CM.sum() - (FP + FN + TP)
-    	
-    return FP, FN, TP, TN
+
+    if binary == True:
+            return FP[1], FN[1], TP[1], TN[1]
+            
+    else: 
+            return FP, FN, TP, TN
+
         
 class Metrics(object):
         
-        def __init__(self, y_pred, y_hat, task, target, threshold = 4):
+        def __init__(self, y_pred, y_hat, target, threshold=4, binary=True):
                 '''Master class from which all the other metrics are computed
                 Arguments
-                y_pred : predicted values
-                y_hat : ground truth
-                task : classification ('class') or regression ('reg')
-                target : irmsd, fnat, class, bin
-                threshold 
+                y_pred:      predicted values
+                y_hat:       ground truth
+                target:      irmsd, fnat, class, bin
+                threshold:   threshold used to split the data into a binary vector
+                binary:      transform the data in binary vectors
                 '''
                 
-                self.task = task
-                self.target = target
                 self.y_pred = y_pred
                 self.y_hat = y_hat
-                
+                self.binary = binary
+                self.target = target
                 self.threshold = threshold
+
                 print ('Threshold set to {}'.format(self.threshold))
                 
-                if self.task == 'reg':
+                if self.binary == True:
                         
                         self.y_pred = get_boolean(self.y_pred, self.threshold, self.target)
                         self.y_hat = get_boolean(self.y_hat, self.threshold, self.target)
-                        
-        def get_metrics(self):            
+                        classes = [0,1]
                 
-                FP, FN, TP, TN = get_comparison(self.y_pred, self.y_hat)
+                else: 
+                        if self.target == 'class':
```

Comment:
maybe the name of that case could be more specific as 'self.target = capri_classes'.

Reply:
I wanted to make it general if one wants to add any new classification, but we can discuss that. So far the capri classification is not implemented into the `get_score()` function of `Graph.py`. That may be the object of the next commit !